### PR TITLE
Merge app

### PR DESCRIPTION
**Pull request by ben-heil**
This PR adds the utility functions and app streamlit app code to visualize the results of the index runs in webapp form

Changes:
- Adds app.py
- Updates environment file with streamlit and plotly (I knew tinkering with plotly this whole project would pay off eventually)
- Adds `get_high_overlap_fields` to subset the network dataframes into only those with enough overlapping papers
- Add convenience functions to `utils` for loading data in a form that is easy to visualize

### CODE REVIEW

File: indices/get_high_overlap_fields.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    i = 0
+    for path in glob.glob(os.path.join(args.in_dir, '*.pkl')):
+        filename = os.path.basename(path)
+        try:
+            with open(path, 'rb') as in_file:
+                df = pkl.load(in_file)
+        except FileNotFoundError:
+            continue
+
+        if len(df) > OVERLAP_THRESHOLD:
```
Comment (jjc2718) [inline indices/get_high_overlap_fields.py]: Does this work? I don't see `OVERLAP_THRESHOLD` defined anywhere - I would have thought you have to use `args.overlap_threshold` here instead

File: indices/utils.py [hunk 2]
```diff
@@ -151,4 +153,148 @@ def parse_mesh_headings(metadata_dir: str,
         dois = set(article_df['doi'])
         heading_to_dois[heading] = dois
 
-    return heading_to_dois
\ No newline at end of file
+    return heading_to_dois
+
+
+def calculate_percentiles(true_vals, doi_to_shuffled_metrics):
```
Comment (jjc2718) [inline indices/utils.py:159]: If I remember correctly some of these functions are in your notebooks too, right? Would it be a lot of work to have a single version of them and call the same functions from the web app and the notebook, or is it not worth it to reorganize your project structure that way?

**Review by jjc2718 — APPROVED** [review]
Looks good!

File: indices/get_high_overlap_fields.py (hunk 1 above)
Comment (ben-heil) [reply indices/get_high_overlap_fields.py]: Good catch, I swore I replaced all the instances but I was wrong

File: indices/utils.py (hunk 2 above)
Comment (ben-heil) [reply indices/utils.py:159]: Right, the functions originally lived in the notebooks, then I copied them over when the web app was its own directory, then I moved them into `utils` when merged the two directories. I will add an issue to use the `utils` version of the functions in whichever notebooks I end up keeping 

https://github.com/greenelab/indices/issues/12

**Pull request merged**